File: backend/src/main/java/com/johansvartdal/SpringAI/model/UserSalgsoppgaveJob.java
package com.johansvartdal.SpringAI.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@Entity
@Table
public class UserSalgsoppgaveJob {

    @Id
    @GeneratedValue(strategy =  GenerationType.UUID)
    private String id;

    @ManyToOne
    @JsonIgnore
    private User user;

    @ManyToOne
    private SalgsoppgaveJob salgsoppgaveJob;

    @ManyToOne
    @JoinColumn(name = "chat_conversation_id")
    private ChatConversation chatConversation;
}

File: frontend/src/types/UserSalgsoppgaveJob.ts
import {User} from "./User";
import {SalgsoppgaveJob} from "./SalgsoppgaveJob";
import {ChatConversation} from "./ChatConversation";

export interface UserSalgsoppgaveJob {
    id: string;
    user: User;
    salgsoppgaveJob: SalgsoppgaveJob
    chatConversation?: ChatConversation
}